repxe_host.rb

```diff
@@ -0,0 +1,325 @@
+#!/usr/bin/env ruby
+#
+# repxe_host.rb
+#
+# This script coordinates the re-installation and re-chefing of an
+# existing host.
+#
+# Provisos:
+#
+#  * It still calls out to c-a-r for the chef bootstrap and re-chefing.
+#  * Hosts still have to be manually rebooted.
+#
+# To use from the command line:
+#
+#  1. If necessary, configure your local rubygems mirror.
+#     Replace 'http://mirror.example.com' with your actual mirror.
+#     ```
+#     bundle config mirror.https://rubygems.org http://mirror.example.com
+#     ```
+#
+#  2. Run 'bundle install --deployment' on a bootstrap node with
+#     access to a rubygems mirror.
+#
+#  3. If not already using the target bootstrap, sync the updated
+#     repository, including 'vendor' directory, to the target
+#     bootstrap.
+#
+#  4. Run 'bundle exec ./repxe_host.rb <hostname>' to begin the process.
+#
+#  5. When prompted, manually reboot the host, then press enter.
+#
+# It is also possible to use methods from this script at a ruby REPL
+# instead of running the script from a UNIX shell.  To load methods
+# into `irb`:
+#
+#  1. Change to the repo directory.
+#
+#  2. Verify that dependencies are installed:
+#     `bundle list`
+#
+#  3. Run irb inside the repo directory.
+#     `bundle exec irb`
+#
+#  4. Load this file.
+#     `irb(main):001:0> load ./repxe_host.rb`
+#
+
+require 'chef/provisioning/transport/ssh'
+require 'mixlib/shellout'
+require 'pry'
+require 'timeout'
+
+def cluster_assign_roles(environment,type,entry=nil)
+  if(type.to_s.downcase != 'basic' &&
+     type.to_s.downcase != 'hadoop')
+    raise "#{type} is not one of 'basic', 'hadoop' !"
+  end
+
+  # We use system() instead of Mixlib::ShellOut specifically so that the
+  # child process re-uses our STDOUT/STDERR.
```

Comment:
I thought one could setup pipes for I/O redirection to allow this to still be tested?